### PR TITLE
Mega menu: Update mobile hover state

### PR DESCRIPTION
**Pull request by anselmbradford**
## Changes

- Left align all items in the mobile menu, including global CTA.
- Add green hover to mobile menu items.
- Convert global CTA link to a native link in the menu.
- Rename menu `_classes` helper to `_content_classes` as works on `o-mega-menu_content`
- Add `o-mega-menu_content-list-group` container.

## Testing

1. Pull branch and build.
2. Visit http://localhost:8000/policy-compliance/ and resize to mobile.
3. Note currently selected item and hover state.
4. Desktop should be unchanged from production.

## Screenshots

Screen Shot 2020-03-13 at 5 26 07 PM

Screen Shot 2020-03-13 at 5 42 43 PM

### CODE REVIEW

**Comment by anselmbradford** [comment]
@contolini

**Comment by anselmbradford** [comment]
Contains https://github.com/cfpb/cfgov-refresh/pull/5587

File: cfgov/unprocessed/css/organisms/mega-menu.less [hunk 1]
```diff
@@ -157,58 +163,61 @@ body {
...
             .cf-icon-svg {
                 display: inline-block;
             }
         }
 
-        // Featured menu content is link items with icons.
-        &-list__featured {
+        &-link,
+        &-overview-link {
+            width: 100%;
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/organisms/mega-menu.less]: This actually makes the desktop links wrap later, so the right-hand edge is too close sometimes I think, but we can adjust in another PR.

File: cfgov/unprocessed/css/organisms/mega-menu.less [hunk 2]
```diff
@@ -360,51 +369,56 @@ body {
                 border-bottom: 1px solid @gray-40;
             }
 
+            &-grid {
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/organisms/mega-menu.less]: The whole `-grid` container we can likely remove, but should be done in another PR.

File: cfgov/unprocessed/css/organisms/mega-menu.less (hunk 1 above)
Comment (anselmbradford) [reply cfgov/unprocessed/css/organisms/mega-menu.less]: Definitely too close

Screen Shot 2020-03-19 at 11 41 26 AM

Comment (contolini) [reply cfgov/unprocessed/css/organisms/mega-menu.less]: There's also an overlap on certain screen sizes.

Screen Shot 2020-03-19 at 1 04 08 PM

File: cfgov/unprocessed/css/organisms/mega-menu.less [hunk 3]
```diff
@@ -414,27 +435,50 @@ body {
...
+                    outline: 1px dotted @black;
+                }
+
+                // This is styling for the currently selected content-2 links.
+                &__current:hover,
+                &__current,
+                &__current:visited {
+                    color: @black;
+                    border-left-color: @black;
+                    cursor: default;
```
Comment (contolini) [inline cfgov/unprocessed/css/organisms/mega-menu.less]: Why is this necessary?

File: cfgov/unprocessed/css/organisms/mega-menu.less (hunk 1 above)
Comment (contolini) [reply cfgov/unprocessed/css/organisms/mega-menu.less]: You could calc it: `width: calc(100% - 5px);`

File: cfgov/jinja2/v1/_includes/organisms/mega-menu.html [hunk 4]
```diff
@@ -108,68 +114,86 @@
...
+                       {{ _content_classes( nav_depth, '-overview-heading' ) }}">
+                <a class="{{ _content_classes( nav_depth, '-overview-link' ) }}
+                            {{ _content_classes( nav_depth, '-overview-link__current' ) if nav_overview_url == request.path else '' }}"
+                            {{ '' if nav_overview_url == request.path or nav_overview_url == '#' else 'href=' + nav_overview_url | e }}>
                     {{ nav_overview_text ~ ' Overview' }}
                 </a>
             </h3>
             {% endif %}
 
+            {# TODO: Make the follow menu markup use the _nav_item macro. #}
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply cfgov/jinja2/v1/_includes/organisms/mega-menu.html]: Fixed! https://github.com/cfpb/cfgov-refresh/pull/5579/commits/5db332393ad1bfaa659fc8a4fd75eff0b4553cc0

File: cfgov/unprocessed/css/organisms/mega-menu.less [hunk 5]
```diff
@@ -360,51 +376,56 @@ body {
...
-
             &-link,
             &-overview-link {
                 position: relative;
                 // Colors for :link, :visited, :hover, :focus, :active.
                 .u-link__colors( @pacific, @pacific, @pacific-60, @pacific, @dark-navy );
 
                 padding-top: unit( @grid_gutter-width / 2 / @base-font-size-px, em );
-                padding-right: unit( @grid_gutter-width / 2 / @base-font-size-px, em );
+                //padding-right: unit( @grid_gutter-width / 2 / @base-font-size-px, em );
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply cfgov/unprocessed/css/organisms/mega-menu.less]: Fixed! https://github.com/cfpb/cfgov-refresh/pull/5579/commits/5db332393ad1bfaa659fc8a4fd75eff0b4553cc0

File: cfgov/unprocessed/css/organisms/mega-menu.less (hunk 3 above)
Comment (anselmbradford) [reply cfgov/unprocessed/css/organisms/mega-menu.less]: It's assuming the current page's link in the menu would not be treated as a link, so it's changing it from the hand to the default pointer cursor.

File: cfgov/unprocessed/css/organisms/mega-menu.less (hunk 1 above)
Comment (anselmbradford) [reply cfgov/unprocessed/css/organisms/mega-menu.less]: Hmm, maybe. We want it 100% in some cases. I'll leave this one to another PR if you don't mind.
